File: src/main/java/com/tombay11/service/BuyerService.java
package com.tombay11.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.tombay11.domain.Buyer;
import com.tombay11.dto.BuyerDTO;
import com.tombay11.dto.BuyerSearchDTO;
import com.tombay11.dto.BuyerPageDTO;
import com.tombay11.dto.BuyerConvertCriteriaDTO;
import com.tombay11.service.GenericService;
import com.tombay11.dto.common.RequestDTO;
import com.tombay11.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface BuyerService extends GenericService<Buyer, Integer> {

	List<Buyer> findAll();

	ResultDTO addBuyer(BuyerDTO buyerDTO, RequestDTO requestDTO);

	ResultDTO updateBuyer(BuyerDTO buyerDTO, RequestDTO requestDTO);

    Page<Buyer> getAllBuyers(Pageable pageable);

    Page<Buyer> getAllBuyers(Specification<Buyer> spec, Pageable pageable);

	ResponseEntity<BuyerPageDTO> getBuyers(BuyerSearchDTO buyerSearchDTO);
	
	List<BuyerDTO> convertBuyersToBuyerDTOs(List<Buyer> buyers, BuyerConvertCriteriaDTO convertCriteria);

	BuyerDTO getBuyerDTOById(Integer buyerId);







}

File: tombay11_ui/src/services/BuyerService.js
import http from "../http-common";

class BuyerService {
  getAllBuyers(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/buyer/buyers`, searchDTO);
  }

  get(buyerId) {
    return this.getRequest(`/buyer/${buyerId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/buyer?field=${matchData}`, null);
  }

  addBuyer(data) {
    return http.post("/buyer/addBuyer", data);
  }

  update(data) {
  	return http.post("/buyer/updateBuyer", data);
  }
  
  uploadImage(data,buyerId) {
  	return http.postForm("/buyer/uploadImage/"+buyerId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new BuyerService();
